--- git-2.24.1/daemon.c	2019-12-10 18:29:01.000000000 +1100
+++ git-2.24.1/daemon.c.orig	2019-12-19 17:35:01.597682504 +1100
@@ -110,7 +110,7 @@
 {
 	va_list params;
 	va_start(params, err);
-	logreport(LOG_ERR, err, params);
+	// logreport(LOG_ERR, err, params);
 	va_end(params);
 }

@@ -121,13 +121,13 @@
 	if (!verbose)
 		return;
 	va_start(params, err);
-	logreport(LOG_INFO, err, params);
+	// logreport(LOG_INFO, err, params);
 	va_end(params);
 }

 static void NORETURN daemon_die(const char *err, va_list params)
 {
-	logreport(LOG_ERR, err, params);
+	// logreport(LOG_ERR, err, params);
 	exit(1);
 }

@@ -1431,12 +1431,12 @@
 			log_destination = LOG_DESTINATION_STDERR;
 	}

-	if (log_destination == LOG_DESTINATION_SYSLOG) {
-		openlog("git-daemon", LOG_PID, LOG_DAEMON);
-		set_die_routine(daemon_die);
-	} else
-		/* avoid splitting a message in the middle */
-		setvbuf(stderr, NULL, _IOFBF, 4096);
+	// if (log_destination == LOG_DESTINATION_SYSLOG) {
+	// 	openlog("git-daemon", LOG_PID, LOG_DAEMON);
+	// 	set_die_routine(daemon_die);
+	// } else
+	// 	/* avoid splitting a message in the middle */
+  setvbuf(stderr, NULL, _IOFBF, 4096);

 	if (inetd_mode && (detach || group_name || user_name))
 		die("--detach, --user and --group are incompatible with --inetd");
