--- git-2.9.5.orig/daemon.c	2017-08-09 21:48:31.000000000 +0200
+++ git-2.9.5/daemon.c	2018-09-11 23:13:41.000000000 +0200
@@ -99,7 +99,7 @@
 {
 	va_list params;
 	va_start(params, err);
-	logreport(LOG_ERR, err, params);
+//	logreport(LOG_ERR, err, params);
 	va_end(params);
 }
 
@@ -110,13 +110,13 @@
 	if (!verbose)
 		return;
 	va_start(params, err);
-	logreport(LOG_INFO, err, params);
+//	logreport(LOG_INFO, err, params);
 	va_end(params);
 }
 
 static void NORETURN daemon_die(const char *err, va_list params)
 {
-	logreport(LOG_ERR, err, params);
+//	logreport(LOG_ERR, err, params);
 	exit(1);
 }
 
@@ -1339,10 +1339,14 @@
 		usage(daemon_usage);
 	}
 
+/*
+	 no syslog on QNX NTO
+
 	if (log_syslog) {
 		openlog("git-daemon", LOG_PID, LOG_DAEMON);
 		set_die_routine(daemon_die);
 	} else
+*/
 		/* avoid splitting a message in the middle */
 		setvbuf(stderr, NULL, _IOFBF, 4096);
 
