--- src/Makefile.in.orig	2020-01-07 07:12:46.056599822 +1100
+++ src/Makefile.in	2019-10-19 08:22:33.000000000 +1100
@@ -112,7 +112,7 @@
 	listbox.$(OBJEXT) llist.$(OBJEXT) note.$(OBJEXT) \
 	notify.$(OBJEXT) pcal.$(OBJEXT) queue.$(OBJEXT) \
 	recur.$(OBJEXT) sha1.$(OBJEXT) sigs.$(OBJEXT) \
+	strings.$(OBJEXT) todo.$(OBJEXT) langinfo.$(OBJEXT) ui-calendar.$(OBJEXT) \
-	strings.$(OBJEXT) todo.$(OBJEXT) ui-calendar.$(OBJEXT) \
 	ui-day.$(OBJEXT) ui-todo.$(OBJEXT) utf8.$(OBJEXT) \
 	utils.$(OBJEXT) vars.$(OBJEXT) vector.$(OBJEXT) wins.$(OBJEXT) \
 	mem.$(OBJEXT) dmon.$(OBJEXT)
