--- gettext-tools/Makefile.in.orig	2010-06-06 14:12:33.000000000 -0600
+++ gettext-tools/Makefile.in	2012-07-01 12:41:11.000000000 -0600
@@ -1248,7 +1248,7 @@
 top_srcdir = @top_srcdir@
 AUTOMAKE_OPTIONS = 1.5 gnu no-dependencies
 ACLOCAL_AMFLAGS = -I m4 -I ../gettext-runtime/m4 -I ../m4 -I gnulib-m4 -I libgrep/gnulib-m4 -I libgettextpo/gnulib-m4
-SUBDIRS = doc intl gnulib-lib libgrep src libgettextpo po projects styles misc man m4 tests gnulib-tests examples
+SUBDIRS = doc intl gnulib-lib libgrep src libgettextpo po projects styles misc man m4 tests gnulib-tests
 
 # Allow users to use "gnulib-tool --update".
 
