--- src/cmds_normal.c.orig	2019-12-10 15:04:51.434781815 +1100
+++ src/cmds_normal.c	2019-12-10 15:07:27.422781815 +1100
@@ -226,35 +226,7 @@
         // CTRL d
         case ctl('d'):                      // set date format using current locate D_FMT format
             {
-        #ifdef USELOCALE
-            #include <locale.h>
-            #include <langinfo.h>
-            char * loc = NULL;
-            char * f = NULL;
-            loc = setlocale(LC_TIME, "");
-            if (loc != NULL) {
-                f = nl_langinfo(D_FMT);
-            } else {
-                sc_error("No locale set. Nothing changed");
-            }
-            int p, r = currow, c = curcol, rf = currow, cf = curcol;
-            if ( (p = is_range_selected()) != -1) {
-                struct srange * sr = get_range_by_pos(p);
-                r = sr->tlrow;
-                c = sr->tlcol;
-                rf = sr->brrow;
-                cf = sr->brcol;
-            }
-            if (any_locked_cells(r, c, rf, cf)) {
-                sc_error("Locked cells encountered. Nothing changed");
-                return;
-            }
-            dateformat(lookat(r, c), lookat(rf, cf), f);
-            ui_update(TRUE);
-            break;
-        #else
             sc_info("Build made without USELOCALE enabled");
-        #endif
             }
 
         // CTRL f
