--- src/cmds_visual.c.orig	2019-12-10 15:05:02.558781815 +1100
+++ src/cmds_visual.c	2019-12-10 15:08:35.270781815 +1100
@@ -399,28 +399,7 @@
 
     // datefmt with locale D_FMT format
     } else if (buf->value == ctl('d')) {
-        #ifdef USELOCALE
-            #include <locale.h>
-            #include <langinfo.h>
-            char * loc = NULL;
-            char * f = NULL;
-            loc = setlocale(LC_TIME, "");
-            if (loc != NULL) {
-                f = nl_langinfo(D_FMT);
-            } else {
-                sc_error("No locale set. Nothing changed");
-            }
-            if (any_locked_cells(r->tlrow, r->tlcol, r->brrow, r->brcol)) {
-                sc_error("Locked cells encountered. Nothing changed");
-                return;
-            }
-            dateformat(lookat(r->tlrow, r->tlcol), lookat(r->brrow, r->brcol), f);
-        exit_visualmode();
-        chg_mode('.');
-        ui_show_header();
-        #else
             sc_info("Build made without USELOCALE enabled");
-        #endif
 
     // EDITION COMMANDS
     // yank
