--- git-2.9.5.orig/configure.ac	2017-08-09 21:48:31.000000000 +0200
+++ git-2.9.5/configure.ac	2018-05-18 01:19:43.000000000 +0200
@@ -856,26 +856,6 @@
 fi
 GIT_CONF_SUBST([NO_REGEX])
 #
-# Define FREAD_READS_DIRECTORIES if your are on a system which succeeds
-# when attempting to read from an fopen'ed directory.
-AC_CACHE_CHECK([whether system succeeds to read fopen'ed directory],
- [ac_cv_fread_reads_directories],
-[
-AC_RUN_IFELSE(
-	[AC_LANG_PROGRAM([AC_INCLUDES_DEFAULT],
-		[[char c;
-		FILE *f = fopen(".", "r");
-		return f && fread(&c, 1, 1, f)]])],
-	[ac_cv_fread_reads_directories=no],
-	[ac_cv_fread_reads_directories=yes])
-])
-if test $ac_cv_fread_reads_directories = yes; then
-	FREAD_READS_DIRECTORIES=UnfortunatelyYes
-else
-	FREAD_READS_DIRECTORIES=
-fi
-GIT_CONF_SUBST([FREAD_READS_DIRECTORIES])
-#
 # Define SNPRINTF_RETURNS_BOGUS if your are on a system which snprintf()
 # or vsnprintf() return -1 instead of number of characters which would
 # have been written to the final string if enough space had been available.
