--- git-2.9.5.orig/configure.ac	2018-05-18 01:27:53.000000000 +0200
+++ git-2.9.5/configure.ac	2018-05-18 01:29:38.000000000 +0200
@@ -855,64 +855,6 @@
 	NO_REGEX=YesPlease
 fi
 GIT_CONF_SUBST([NO_REGEX])
-#
-# Define SNPRINTF_RETURNS_BOGUS if your are on a system which snprintf()
-# or vsnprintf() return -1 instead of number of characters which would
-# have been written to the final string if enough space had been available.
-AC_CACHE_CHECK([whether snprintf() and/or vsnprintf() return bogus value],
- [ac_cv_snprintf_returns_bogus],
-[
-AC_RUN_IFELSE(
-	[AC_LANG_PROGRAM([AC_INCLUDES_DEFAULT
-		#include "stdarg.h"
-
-		int test_vsnprintf(char *str, size_t maxsize, const char *format, ...)
-		{
-		  int ret;
-		  va_list ap;
-		  va_start(ap, format);
-		  ret = vsnprintf(str, maxsize, format, ap);
-		  va_end(ap);
-		  return ret;
-		}],
-		[[char buf[6];
-		  if (test_vsnprintf(buf, 3, "%s", "12345") != 5
-		      || strcmp(buf, "12")) return 1;
-		  if (snprintf(buf, 3, "%s", "12345") != 5
-		      || strcmp(buf, "12")) return 1]])],
-	[ac_cv_snprintf_returns_bogus=no],
-	[ac_cv_snprintf_returns_bogus=yes])
-])
-if test $ac_cv_snprintf_returns_bogus = yes; then
-	SNPRINTF_RETURNS_BOGUS=UnfortunatelyYes
-else
-	SNPRINTF_RETURNS_BOGUS=
-fi
-GIT_CONF_SUBST([SNPRINTF_RETURNS_BOGUS])
-#
-# Define NEEDS_MODE_TRANSLATION if your OS strays from the typical file type
-# bits in mode values.
-AC_CACHE_CHECK([whether the platform uses typical file type bits],
- [ac_cv_sane_mode_bits], [
-AC_EGREP_CPP(yippeeyeswehaveit,
-	AC_LANG_PROGRAM([AC_INCLUDES_DEFAULT],
-[#if S_IFMT == 0170000 && \
-	S_IFREG == 0100000 && S_IFDIR == 0040000 && S_IFLNK == 0120000 && \
-	S_IFBLK == 0060000 && S_IFCHR == 0020000 && \
-	S_IFIFO == 0010000 && S_IFSOCK == 0140000
-yippeeyeswehaveit
-#endif
-]),
-	[ac_cv_sane_mode_bits=yes],
-	[ac_cv_sane_mode_bits=no])
-])
-if test $ac_cv_sane_mode_bits = yes; then
-	NEEDS_MODE_TRANSLATION=
-else
-	NEEDS_MODE_TRANSLATION=UnfortunatelyYes
-fi
-GIT_CONF_SUBST([NEEDS_MODE_TRANSLATION])
-
 
 ## Checks for library functions.
 ## (in default C library and libraries checked by AC_CHECK_LIB)
