--- src/list.c.orig	2012-06-27 08:44:49.000000000 -0600
+++ src/list.c	2012-06-27 09:07:49.000000000 -0600
@@ -25,7 +25,9 @@
 
 #include "common.h"
 
+#if !defined(max)
 #define max(a, b) ((a) < (b) ? (b) : (a))
+# endif
 
 union block *current_header;	/* points to current archive header */
 enum archive_format current_format; /* recognized format */
@@ -647,7 +649,7 @@
 	case BLKTYPE:
 	case CHRTYPE:
 	  stat_info->stat.st_rdev =
-	    makedev (MAJOR_FROM_HEADER (header->header.devmajor),
+	    makedev (0,MAJOR_FROM_HEADER (header->header.devmajor),
 		     MINOR_FROM_HEADER (header->header.devminor));
 	  break;
 
