--- groff-1.22.3/doc/Makefile.sub.orig	2014-12-25 23:01:32.844127487 -0500
+++ groff-1.22.3/doc/Makefile.sub	2014-12-25 23:02:21.966227789 -0500
@@ -17,8 +17,8 @@
 # along with this program. If not, see <http://www.gnu.org/licenses/>.
 
 # These may be overridden if cross-compiling.
-GROFFBIN=$(top_builddir)/src/roff/groff/groff
-GROFF_BIN_PATH=`echo $(groff_bin_dirs) | sed -e 's|  *|$(SH_SEP)|g'`
+GROFFBIN=/usr/bin/groff
+GROFF_BIN_PATH=/usr/bin
 
 groff_bin_dirs=\
   $(top_builddir)/src/roff/groff \
