--- groff-1.22.3/doc/Makefile.in.orig	2014-12-25 22:56:40.651650854 -0500
+++ groff-1.22.3/doc/Makefile.in	2014-12-25 22:59:17.090330056 -0500
@@ -34,10 +34,9 @@
 pnmtops=@pnmtops_nosetpage@
 
 # These may be overridden if cross-compiling.
-TROFFBIN=$(top_builddir)/src/roff/troff/troff
-GROFFBIN=$(top_builddir)/src/roff/groff/groff
-GROFF_BIN_PATH=`echo $(groff_bin_dirs) \
-                | LANG=C LC_ALL=C sed -e 's|  *|:|g'`
+TROFFBIN=/usr/bin/troff
+GROFFBIN=/usr/bin/groff
+GROFF_BIN_PATH=/usr/bin
 
 # info files are distributed with the groff package, so makeinfo is
 # usually not needed (however, to build from the repo you NEED it!)
