--- groff-1.22.3/configure.orig	2014-12-25 23:13:25.630562852 -0500
+++ groff-1.22.3/configure	2014-12-25 23:13:58.487965049 -0500
@@ -6728,11 +6728,11 @@
 
    { $as_echo "$as_me:${as_lineno-$LINENO}: result: $groff_cv_decl_needed_hypot" >&5
 $as_echo "$groff_cv_decl_needed_hypot" >&6; }
-   if test $groff_cv_decl_needed_hypot = yes; then
-
-$as_echo "#define NEED_DECLARATION_HYPOT 1" >>confdefs.h
-
-   fi
+#   if test $groff_cv_decl_needed_hypot = yes; then
+#
+#$as_echo "#define NEED_DECLARATION_HYPOT 1" >>confdefs.h
+#
+#   fi
    ac_ext=c
 ac_cpp='$CPP $CPPFLAGS'
 ac_compile='$CC -c $CFLAGS $CPPFLAGS conftest.$ac_ext >&5'
