--- groff-1.22.3/contrib/mom/Makefile.sub.orig	2014-12-25 22:50:25.867650309 -0500
+++ groff-1.22.3/contrib/mom/Makefile.sub	2014-12-25 22:53:39.319905273 -0500
@@ -18,9 +18,9 @@
 # along with this program. If not, see <http://www.gnu.org/licenses/>.
 
 # These may be overridden if cross-compiling.
-GROFFBIN=$(top_builddir)/src/roff/groff/groff
-GROFF_BIN_PATH=`echo $(groff_bin_dirs) | sed -e 's|  *|$(SH_SEP)|g'`
-PDFMOMBIN=$(top_builddir)/src/devices/gropdf/pdfmom
+GROFFBIN=/usr/bin/groff
+GROFF_BIN_PATH=/usr/bin
+PDFMOMBIN=/usr/bin/pdfmom
 
 groff_bin_dirs=\
   $(top_builddir)/src/roff/groff \
