--- gcc/libstdc++-v3/include/Makefile.am.orig	2014-09-28 00:35:08.000000000 +0400
+++ gcc/libstdc++-v3/include/Makefile.am	2014-09-28 01:11:05.000000000 +0400
@@ -690,7 +690,6 @@ c_compatibility_headers = \
 	${c_compatibility_srcdir}/stdint.h \
 	${c_compatibility_srcdir}/stdio.h \
 	${c_compatibility_srcdir}/stdlib.h \
-	${c_compatibility_srcdir}/string.h \
 	${c_compatibility_srcdir}/tgmath.h \
 	${c_compatibility_srcdir}/time.h \
 	${c_compatibility_srcdir}/wchar.h \
--- gcc/libstdc++-v3/include/Makefile.in.orig	2014-09-28 00:35:14.000000000 +0400
+++ gcc/libstdc++-v3/include/Makefile.in	2014-09-28 01:11:31.000000000 +0400
@@ -934,7 +934,6 @@ c_compatibility_builddir = .
 @GLIBCXX_C_HEADERS_C_TRUE@	${c_compatibility_srcdir}/stdint.h \
 @GLIBCXX_C_HEADERS_C_TRUE@	${c_compatibility_srcdir}/stdio.h \
 @GLIBCXX_C_HEADERS_C_TRUE@	${c_compatibility_srcdir}/stdlib.h \
-@GLIBCXX_C_HEADERS_C_TRUE@	${c_compatibility_srcdir}/string.h \
 @GLIBCXX_C_HEADERS_C_TRUE@	${c_compatibility_srcdir}/tgmath.h \
 @GLIBCXX_C_HEADERS_C_TRUE@	${c_compatibility_srcdir}/time.h \
 @GLIBCXX_C_HEADERS_C_TRUE@	${c_compatibility_srcdir}/wchar.h \
