--- gcc/binutils/gas/configure.tgt.orig	2012-07-21 21:37:33.000000000 -0400
+++ gcc/binutils/gas/configure.tgt	2012-07-21 20:57:57.000000000 -0400
@@ -118,7 +118,7 @@
   arm-*-uclinux*)			fmt=elf  em=linux ;;
   arm-*-netbsdelf*)                 	fmt=elf  em=nbsd ;;
   arm-*-*n*bsd*)			fmt=aout em=nbsd ;;
-  arm-*-nto*)				fmt=elf ;;
+  arm-*-nto*)				fmt=elf need_libm=yes ;;
   arm-epoc-pe | thumb-epoc-pe)		fmt=coff em=epoc-pe ;;
   arm-wince-pe | arm-*-wince)		fmt=coff em=wince-pe ;;
   arm-*-pe | thumb-*-pe)		fmt=coff em=pe ;;
