diff --git default_options.h default_options.h
index 9000fcc..2f796c4 100644
--- default_options.h
+++ default_options.h
@@ -19,9 +19,9 @@ IMPORTANT: Some options will require "make clean" after changes */
 #define DROPBEAR_DEFADDRESS ""
 
 /* Default hostkey paths - these can be specified on the command line */
-#define DSS_PRIV_FILENAME "/etc/dropbear/dropbear_dss_host_key"
-#define RSA_PRIV_FILENAME "/etc/dropbear/dropbear_rsa_host_key"
-#define ECDSA_PRIV_FILENAME "/etc/dropbear/dropbear_ecdsa_host_key"
+#define DSS_PRIV_FILENAME "/accounts/1000/shared/documents/clitools/etc/dropbear/dropbear_dss_host_key"
+#define RSA_PRIV_FILENAME "/accounts/1000/shared/documents/clitools/etc/dropbear/dropbear_rsa_host_key"
+#define ECDSA_PRIV_FILENAME "/accounts/1000/shared/documents/clitools/etc/dropbear/dropbear_ecdsa_host_key"
 
 /* Set NON_INETD_MODE if you require daemon functionality (ie Dropbear listens
  * on chosen ports and keeps accepting connections. This is the default.
diff --git dropbearconvert.c dropbearconvert.c
index 7285f23..39b7454 100644
--- dropbearconvert.c
+++ dropbearconvert.c
@@ -49,7 +49,7 @@ static void printhelp(char * progname) {
 					"dropbear\n"
 					"\n"
 					"Example:\n"
-					"dropbearconvert openssh dropbear /etc/ssh/ssh_host_rsa_key /etc/dropbear_rsa_host_key\n",
+					"dropbearconvert openssh dropbear ~/etc/ssh/ssh_host_rsa_key ~/etc/dropbear_rsa_host_key\n",
 					progname);
 }
 
