--- configure.orig	2012-07-01 16:41:51.000000000 -0600
+++ configure	2012-07-01 16:48:02.000000000 -0600
@@ -38926,13 +38926,13 @@
   esac
 fi
 
-if test -z "$ac_list_mounted_fs"; then
-  { { echo "$as_me:$LINENO: error: could not determine how to read list of mounted file systems" >&5
-echo "$as_me: error: could not determine how to read list of mounted file systems" >&2;}
-   { (exit 1); exit 1; }; }
-  # FIXME -- no need to abort building the whole package
-  # Can't build mountlist.c or anything that needs its functions
-fi
+# if test -z "$ac_list_mounted_fs"; then
+#   { { echo "$as_me:$LINENO: error: could not determine how to read list of mounted file systems" >&5
+# echo "$as_me: error: could not determine how to read list of mounted file systems" >&2;}
+#    { (exit 1); exit 1; }; }
+#   # FIXME -- no need to abort building the whole package
+#   # Can't build mountlist.c or anything that needs its functions
+# fi
 
 if test $ac_list_mounted_fs = found; then
   gl_cv_list_mounted_fs=yes
